spec/integration/catch_errors_spec.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt_spec/config'
+require 'bolt_spec/conn'
+require 'bolt_spec/files'
+require 'bolt_spec/integration'
+require 'bolt/util'
+
+describe "catch_errors" do
+  include BoltSpec::Integration
+  include BoltSpec::Config
+  include BoltSpec::Conn
+
+  after(:each) { Puppet.settings.send(:clear_everything_for_tests) }
+
+  let(:modulepath) { [fixture_path('modules'), fixture_path('apply')].join(File::PATH_SEPARATOR) }
+  let(:target) do
```

Comment:
I think that the tests cover the right cases. I get that our current integration test setup can safely assume that the bolt controller being Linux signals that SSH targets are available (If we are calling OMI linux) and that if it is windows winrm ones are. However I think we should get to a place where the bolt controller OS is not used to decide what transport to use to act on targets (with the obvious caveat of local transport). So in this case I think you could lay down some inventory data (which it looks like we might allready have handy https://github.com/puppetlabs/bolt/blob/ea5b61a8daf363d679ed21225564dfb4fab6ad79/spec/lib/bolt_spec/conn.rb#L54-L63)  and run tests against "all", then iterate over the results. This approach would also avoid having both winrm and ssh specific flags in the `config_flags` helper.

Reply:
Though I guess the ssh nodes would still fail on appveyor and vice versa with travis. So I guess that leaves just using local transport and a ruby task?